python/cudf/dataframe/index.py

```diff
@@ -537,6 +561,26 @@ def find_label_range(self, first, last):
             end += 1
         return begin, end
 
+    @property
+    def is_unique(self):
+        return len(set(self._values)) == len(self._values)
```

Comment:
You can't use `set(...)` here as that doesn't know how to operate against gpu values. You'd likely want `.nunique` for a Series and `.unique_count` for a column.